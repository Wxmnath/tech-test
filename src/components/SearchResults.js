import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchResults.css";

function SearchResults({ results }) {
  if (!results.length) {
    return <p className="no-result-title">No results found!</p>;
  } else {
    return (
      <>
        {results.map((image) => (
          <img
            className="search-results"
            src={image}
            aria-hidden
            alt="space-image"
          />
        ))}
      </>
    );
  }
}

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string.isRequired),
};

export default SearchResults;
