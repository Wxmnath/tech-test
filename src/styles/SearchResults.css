.search-results {
  border: 3px solid black;
  border-radius: 11px;
  max-width: 100%;
  height: auto;
  margin-bottom: 50px;
}

.search-result {
  color: darkblue;
  font-size: 1.3rem;
  font-weight: bold;
  font-style: oblique;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
}
