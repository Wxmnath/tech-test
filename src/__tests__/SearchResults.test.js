import { render } from "@testing-library/react";
import SearchResults from "../components/Search";

describe("Search Result", () => {
  test("renders search result", () => {
    render(<SearchResults />);
    const { asFragment } = render(<SearchResults />);
    expect(asFragment).toMatchSnapshot();
  });
});
