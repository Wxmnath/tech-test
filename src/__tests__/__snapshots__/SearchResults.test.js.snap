// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result renders search result 1`] = `[Function]`;
